src/cluster_asm.c

```diff
@@ -1894,6 +2049,35 @@ void clusterSyncSlotsCommand(client *c) {
 
                 sdsfreesplitres(parts, count);
                 addReply(c, shared.ok);
+            } else if (!strcasecmp(c->argv[j]->ptr, "master-task")) {
+                /* master-task task_id:source_node:dest_node:operation:state:slot_ranges */
+                if (clusterNodeIsMaster(getMyClusterNode())) {
+                    addReplyError(c, "CLUSTER SYNCSLOTS CONF MASTER-TASK only allowed on replica");
+                    return;
+                }
+                int send_event = 0;
+                sds task_info = c->argv[j + 1]->ptr;
+                asmTask *task = asmTaskDeserialize(task_info);
+                if (task) {
+                    int event = asmTaskStateToEvent(task);
+                    if (asmManager->master_task) {
+                        /* Notify when the task is changed, to avoid duplicated notification. */
+                        if (strcmp(task->id, asmManager->master_task->id) != 0 ||
+                            task->operation != asmManager->master_task->operation ||
+                            event != asmTaskStateToEvent(asmManager->master_task))
+                        {
+                            send_event = 1;
```

Comment:
If we've sent IMPORT_STARTED event, we need to be sure that the next event is either IMPORT_COMPLETED or IMPORT_FAILED. Otherwise, it may break something in modules. We don't guarantee it here, no?

Reply:
okay, I misinterpreted the code.